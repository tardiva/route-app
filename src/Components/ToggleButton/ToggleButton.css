.toggle-button {
    margin-left: -2px;
    padding: 0;
    width: 50px;
    height: 48px;
    position: absolute;
    top: 16px;
    left: 0;
    color: #282c34;
    border-radius: 2px;
    background: white;
    border: 0;
    box-shadow: 0 0 8px rgba(0,0,0,0.3);
    cursor: pointer;
}