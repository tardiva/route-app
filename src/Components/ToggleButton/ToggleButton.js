import React, { Component } from 'react';
import './ToggleButton.css';
import Icon from 'react-icons-kit';
import {ic_keyboard_arrow_right} from 'react-icons-kit/md/ic_keyboard_arrow_right';

class ToggleButton extends Component {

  render() {
    return (
      <button className="toggle-button" onClick={this.props.handleClick}>
        <Icon icon={ic_keyboard_arrow_right} size="24"/>
      </button>
    );
  }
}

export default ToggleButton;