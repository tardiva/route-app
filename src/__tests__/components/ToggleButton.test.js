import React from 'react';
import { shallow } from 'enzyme';
import ToggleButton from '../../Components/ToggleButton/ToggleButton';

it('should render ToggleButton correctly', () => {
  const wrapper = shallow(<ToggleButton />);
  expect(wrapper).toMatchSnapshot();
});

it('should call handleClick prop', () => {
  const onClickSpy = jest.fn();
  const wrapper = shallow(<ToggleButton handleClick={onClickSpy}/>);
  const button = wrapper.find('.toggle-button');
  button.simulate('click');
  expect(onClickSpy).toHaveBeenCalledTimes(1);
});