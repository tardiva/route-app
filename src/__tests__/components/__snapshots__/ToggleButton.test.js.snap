// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ToggleButton correctly 1`] = `
<button
  className="toggle-button"
>
  <Icon
    fill="currentColor"
    icon={
      Object {
        "children": Array [
          Object {
            "attribs": Object {
              "d": "M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z",
            },
            "name": "path",
          },
        ],
        "viewBox": "0 0 24 24",
      }
    }
    size="24"
  />
</button>
`;
